### PR TITLE
feat(rest): listing license clearing info of a project.

### PR DESCRIPTION
**Pull request by rudra-superrr**
Issue: #2030 

Description: GET request will get all releases of license clearing.

### CODE REVIEW

**Comment by nikkuma7** [comment]
Test successful.
image

**Comment by hoangnt2** [comment]
@rudra-superrr  Screenshot from 2023-07-19 16-53-32
There is an nested list in sw360:releases list. Could you remove it?

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 1]
```diff
@@ -329,6 +331,34 @@ public ResponseEntity<CollectionModel<EntityModel<Project>>> getProjectsFiltered
...
+
+		final Set<String> releaseIds = projectService.getReleaseIds(id, sw360User, transitive);
+		List<Release> releases = releaseIds.stream().map(relId -> wrapTException(() -> {
+			final Release sw360Release = releaseService.getReleaseForUserById(relId, sw360User);
+			releaseService.setComponentDependentFieldsInRelease(sw360Release, sw360User);
+			return sw360Release;
+		})).collect(Collectors.toList());
+
+		List<EntityModel<Release>> releaseList = releases.stream().map(sw360Release -> wrapTException(() -> {
+			final Release embeddedRelease = restControllerHelper.convertToEmbeddedLinkedRelease(sw360Release);
```
[the remark this answers is not in the document]
Comment (rudra-superrr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: done

**Comment by rudra-superrr** [comment]
Hi @hoangnt2 , made the changes as requested.
Kindly verify it once.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 2]
```diff
@@ -1122,6 +1147,24 @@ public RepositoryLinksResource process(RepositoryLinksResource resource) {
         return resource;
     }
 
+    private HalResource<Project> createHalLicenseClearing(Project sw360Project, User sw360User, List<EntityModel<Release>> releases)
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:1150]: Unused sw360User, TException is never thrown

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java [hunk 3]
```diff
@@ -684,6 +694,24 @@ public License convertToEmbeddedLicense(String licenseId) {
...
+        embeddedRelease.setOtherLicenseIds(release.getOtherLicenseIds());
+        embeddedRelease.setName(release.getName());
+        embeddedRelease.setComponentType(release.getComponentType());
+        embeddedRelease.setVersion(release.getVersion());
+        embeddedRelease.setType(null);
+        return embeddedRelease;
+    }
+
+    public void addEmbeddedProjectReleases(HalResource halResource, List<EntityModel<Release>> releases) {
+        halResource.addEmbeddedResource("sw360:release", releases);
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java:712]: Keep `sw360:release` if you can

Comment (rudra-superrr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java:776]: It's already sw360:release.

**Comment by rudra-superrr** [comment]
Review comments resolved.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java (hunk 3 above)
Comment (hoangnt2) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java:777]: @rudra-superrr Oh, sorry I mean `sw360:releases`

Comment (rudra-superrr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java:777]: If I keep sw360:releases, then nested list will come in response structure.

image

Comment (hoangnt2) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java:777]: > If I keep sw360:releases, then nested list will come in response structure.
> 
> image

@rudra-superrr, I got it, thanks

**Comment by hoangnt2** [comment]
Code look good

**Comment by ag4ums** [comment]
closes #2030

**Pull request merged**